pkg/reconciler/dynamicrestmapper/defaultrestmapper_mutable.go

```diff
@@ -41,6 +43,24 @@ func (m *DefaultRESTMapper) add(typeMeta typeMeta) {
 
 	m.kindToPluralResource[kind] = plural
 	m.kindToScope[kind] = meta.RESTScopeRoot
+
+	foundDefaultVersion := false
+	for i := range m.defaultGroupVersions {
+		if m.defaultGroupVersions[i].Group == typeMeta.Group {
+			if typeMeta.Version > m.defaultGroupVersions[i].Version {
```

Comment:
How does this work? Do we have served=true here for type? Not quite sure why highest wins here.

Reply:
Yes, by the time the type info gets here, it was already checked by the controllers:

* BuiltinTypesController:
  * https://github.com/gman0/kcp/blob/drm-builtin-types/pkg/reconciler/dynamicrestmapper/dynamicrestmapper_builtin_controller.go#L176-L184
* DynamicTypesController:
  * https://github.com/gman0/kcp/blob/drm-builtin-types/pkg/reconciler/dynamicrestmapper/dynamicrestmapper_dynamic_controller.go#L403-L411
  * https://github.com/gman0/kcp/blob/drm-builtin-types/pkg/reconciler/dynamicrestmapper/dynamicrestmapper_dynamic_controller.go#L436-L447

The version must be `Served=true` to be added to the `gvkrs` slice, and only this `gvkrs` slice is then added to the mapper.